Userland/Utilities/strings.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+#define DEFAULT_MINIMUM_STRING_LENGTH 4
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    Vector<StringView> paths;
+    size_t minimum_string_length = DEFAULT_MINIMUM_STRING_LENGTH;
+
+    enum StringOffsetFormat {
+        None = 0,
+        Decimal,
+        Octal,
+        Hexadecimal
+    };
+
+    size_t current_fd_position;
+    size_t current_path_position = 0;
+    bool show_paths = false;
+    StringView string_offset_format_sv;
+    size_t string_offset_format = StringOffsetFormat::None;
+    size_t string_offset_position;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(minimum_string_length, "Specify the minimum string length.", nullptr, 'n', "number");
+    args_parser.add_option(show_paths, "Display the path for each matched file.", nullptr, 'p');
+    args_parser.add_option(string_offset_format_sv, "Write offset relative to start of each file in (d)ec, (o)ct, or (h)ex format.", nullptr, 't', "format");
+    args_parser.set_general_help("Write the sequences of printable characters in files or pipes to stdout.");
+    args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    if (minimum_string_length < 1) {
+        warnln("Invalid minimum string length {}", minimum_string_length);
+        return 1;
+    }
+
+    if (string_offset_format_sv == "d") {
+        string_offset_format = StringOffsetFormat::Decimal;
+    } else if (string_offset_format_sv == "o") {
+        string_offset_format = StringOffsetFormat::Octal;
+    } else if (string_offset_format_sv == "h") {
+        string_offset_format = StringOffsetFormat::Hexadecimal;
+    } else {
+        if (!string_offset_format_sv.is_empty()) {
+            warnln("Invalid format {}", string_offset_format_sv);
+            return 1;
+        }
+    }
```

Comment:
This can be put into a [custom accept function for the ArgsParser](https://github.com/SerenityOS/serenity/blob/master/Userland/Utilities/cp.cpp#L28-L58) (you obviously don't need to do as much as the linked example), that feels a bit nicer to use, and we save on one variable that we need to care about.

Reply:
Added in [679d363](https://github.com/SerenityOS/serenity/pull/16013/commits/679d363ed3c2e4da723ff1cc18a18563fc9cc254), I think this should be ok?